prusti-tests/tests/cargotest.rs

```diff
@@ -0,0 +1,152 @@
+
+[ERROR] could not compile `foo`
+
+To learn more, run the command again with --verbose.
+",
+        )
+        .run();
+}
+
+/// Test `cargo-prusti` on one of the crates in `test/cargo_verify`.
+///
+/// Special files and folders in the root of the test crate:
+/// * `output.stdout` and `output.stderr`: if present, they are used to check the output of
+///   `cargo-prusti`.
+/// * `prusti-contracts` and related Prusti crates: during the test they will link to the
+///   corresponding Prusti crate.
+///
+/// For more details on the special syntax allowed in the `output.*` files, check the documentation
+/// of `cargo_test_support`: <https://doc.crates.io/contrib/tests/writing.html>.
+fn test_local_project<T: Into<PathBuf>>(project_name: T) {
+    let mut project_builder = project().no_manifest();
+    let relative_project_path = Path::new("tests/cargo_verify").join(project_name.into());
+    let project_path = fs::canonicalize(&relative_project_path).expect(
+        &format!("Failed to canonicalize the path {:?}", relative_project_path)
+    );
+
+    // Populate the test project with symlinks to the local project
+    let project_path_content = fs::read_dir(&project_path)
+        .expect(&format!("Failed to read directory {:?}", project_path));
+    for entry in project_path_content {
+        let entry = entry.expect(&format!("Failed to read content of {:?}", project_path));
+        let path = entry.path();
+        let file_name = path.as_path().file_name()
+            .expect(&format!("Failed to obtain the name of {:?}", path));
+
+        eprintln!("Creating {:?} -> {:?}", file_name, path);
+        if path.is_dir() {
+            project_builder = project_builder.symlink_dir(path.as_path(), &Path::new(file_name));
+        } else {
+            project_builder = project_builder.symlink(path.as_path(), &Path::new(file_name));
+        }
+    }
+
+    // Create a special symlink for prusti_contract and related Prusti crates
+    let prusti_dev_path = project_path.join("../../../..");
+    let prusti_contract_deps = [
+        "prusti-specs",
+        "prusti-contracts",
+        "prusti-contracts-impl",
+        "prusti-contracts-internal",
+    ];
+    for crate_name in &prusti_contract_deps {
+        project_builder = project_builder.symlink_dir(
+            prusti_dev_path.join(crate_name).as_path(),
+            &Path::new(crate_name)
+        );
+    }
+
+    // Set the expected exit status, stdout and stderr
+    let mut test_builder = project_builder.build().process(cargo_prusti_path());
+    let opt_expected_stdout = fs::read_to_string(project_path.join("output.stdout")).ok();
+    let opt_expected_stderr = fs::read_to_string(project_path.join("output.stderr")).ok();
+    if let Some(ref expected_stdout) = opt_expected_stdout {
+        test_builder.with_stdout(expected_stdout);
+    }
+    if let Some(ref expected_stderr) = opt_expected_stderr {
+        test_builder.with_status(101).with_stderr(expected_stderr);
+    }
+
```

Comment:
Shouldn't we also clean up the symlinks?

Reply:
They are created in a temporary folder (e.g. `prusti-dev/target/debug/tmp/cit/t2/`) which is automatically deleted before running the test a second time.

Reply:
The symlinks are created during the `test_builder.run()` below. The statements above are just declarations.